testlib/src/test/java/com/diffplug/spotless/protobuf/BufStepTest.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.protobuf;
+
+import java.io.File;
+
+import org.junit.jupiter.api.Test;
+
+import com.diffplug.spotless.StepHarnessWithFile;
+import com.diffplug.spotless.tag.BufTest;
+
+@BufTest
+class BufStepTest {
+	@Test
+	void test() throws Exception {
+		try (StepHarnessWithFile harness = StepHarnessWithFile.forStep(BufStep.withVersion(BufStep.defaultVersion()).create())) {
+			harness.testResource(new File("src/main/resources/protobuf/buf/buf.proto"), "protobuf/buf/buf.proto", "protobuf/buf/buf.proto.clean");
```

Comment:
It doesn't really matter, but it should be okay for this to be `new File("foo.proto")`. If `BufStep` is implemented correctly (looks to me like it is), then the non-existent "foo.proto" isn't actually used to read any content from. The only significance of a formatter that takes the actual filename should be that formatting depends on the filename, such as `package-info.java` or something like that.

So it should be fine if "foo.proto" doesn't actually exist, because it's just a fake filename to pass around.

Reply:
I thought that might be the case, but I run into issues:
```
java.lang.RuntimeException: > arguments: [/usr/local/bin/buf, format, /Users/andrewparmet/toast/git-repos/spotless/testlib/foo.proto]
> exit code: 1
>    stdout: (empty)
>    stderr: Failure: input /Users/andrewparmet/toast/git-repos/spotless/testlib/foo.proto was not found - is the directory containing this file defined in your buf.work.yaml?
```